c7n/resources/cloudtrail.py

```diff
@@ -119,6 +119,129 @@ def process(self, resources, event=None):
     def __call__(self, r):
         return self.match(r[self.annotation_key])
 
+@CloudTrail.filter_registry.register('log-metric-filter-pattern')
+class LogMetricFilterPattern(Filter):
+
+    """
+    If pattern entered is found then resource will pass if not then resource will fail
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: log-metric-filter-pattern
+            resource: aws.cloudtrail
+            filters:
+            - type: log-metric-filter-pattern
+                required_pattern: "{ ($.eventName = \"ConsoleLogin\") && ($.additionalEventData.MFAUsed != \"Yes\") }"
+    """
+    schema = type_schema('log-metric-filter-pattern',
+                            required_pattern = {'type':'string'},
+                            required= ['required_pattern'])
+
+    schema_alias = False
+    permissions = ('cloudtrail:DescribeTrails', 'cloudtrail:GetTrailStatus', 'cloudtrail:GetEventSelectors',
+                   'cloudwatch:DescribeAlarms', 'logs:DescribeMetricFilters', 'sns:ListSubscriptions')
+    annotation_key = 'c7n:CheckForMultiTrailMfaLogin'
```

Comment:
this doesn't look right, should probably be 

```suggestion
    annotation_key = 'c7n:LogMetricFilter'
```